### DISCUSSION TITLE
Enable user to update their email in auth.users

### DISCUSSION THREAD

**Discussion by fadethegap** [supabase/supabase#10370] [Questions] [answered]
Many of our subscribers are asking to update their login email address. 

In the [supabase documentation](https://supabase.com/docs/reference/dart/auth-updateuser#update-the-email-for-an-authenticated-user) it says to use the updateUser function.

I'm not familiar with the "final" syntax. Would I run this in an api with the supabase service key?

Can someone share an example?

**Comment by GaryAustin1** [comment]
If you have the user signed in then you just need to have a screen for them to fill in the new email and call:
https://supabase.com/docs/reference/javascript/auth-updateuser
`const { data, error } = await supabase.auth.updateUser({email: 'new@email.com'})`

**Reply by jgturner** [reply]
I did this and it only updated the new_email in the auth user table and does not update the login email.

**Comment by fadethegap** [comment, marked as answer]
Thank you @GaryAustin1. I don't know how I missed that in the docs. I was on some other page.

I was so pumped to implement it, and I've run into the below error. Any idea what's going on?

```import { supabase } from '../utils/supabase'

export default function ChangeEmail() {

  console.table({ supabase })
  async function updateEmail() {
    const { data, error } = await supabase.auth.update.user({
      email: 'test@test.com',
    })
    if (error) {
      console.error(error)
    } else {
      console.log(data)
    }
  }
  return (
    <div
      onClick={updateEmail}
      className="w-64 rounded border-2 bg-slate-400 p-3 text-center"
    >
      Change email
    </div>
  )
}
```

![Screen Shot 2022-11-17 at 12 37 03 PM](https://user-images.githubusercontent.com/47059192/202518429-ef93682c-7369-41ad-a780-2e792b498f65.png)

Additionally, the intellisense is not showing updateUser().

![Screen Shot 2022-11-17 at 12 41 41 PM](https://user-images.githubusercontent.com/47059192/202519117-fe7c1294-c0ac-46e5-a79e-f09592b4b6f2.png)

**Reply by GaryAustin1** [reply]
I'm going to guess you are on v1 supabase-js.  With that is is just .update().

**Reply by fadethegap** [reply]
Yessir, you are correct.

Are there any breaking changes if I update to v2?

**Reply by GaryAustin1** [reply]
Yes.  See this https://supabase.com/docs/reference/javascript/upgrade-guide and release notes in same area of the docs.  You can toggle the docs for v1/v2 up at the top of the screen when you are in javascript section.

**Reply by fadethegap** [reply]
Thank you @GaryAustin1.

I always appreciate your quick response and concise help.
